### PR TITLE
Bug 1497766 - Adding ablity to specify keeping namespace alive

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Describe what this PR does and why we need it**:
Keep namespace alive based on new configuration values
Changes proposed in this pull request
 - Adds new Config options
 - will keep namespace if needed 

**Does this PR depend on another PR (Use this to track when PRs should be merged)**
depends-on <PR>
https://github.com/openshift/ansible-service-broker/pull/464
CATASB PR: https://github.com/fusor/catasb/pull/155
**Which issue this PR fixes (This will close that issue when PR gets merged)**
fixes <issue_number>

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -159,6 +159,8 @@ registry:
 | bearer_token_file | Location of bearer token to be used | N |
 | image_pull_policy | When to pull the image | Y |
 | sandbox_role | Role to give to apb sandbox environment | Y |
+| keep_namespace | Always keep namespace after apb execution | N |
+| Keep_namespace_on_error | Keep namespace after apb execution has an error | N |
```
Comment (jmrodri) [inline docs/config.md]: entry should be lowercase `Keep_namespace_on_error` -> `keep_namespace_on_error`

File: docs/deployment.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ REGISTRY_URL | docker.io | Registry URL
 DEV_BROKER | true | Include Broker Development Endpoint (true/false) 
 AUTO_ESCALATE | true | Auto escalate the users permissions when running an APB 
 BROKER_CA_CERT | "" | Tells the broker that the ca that has signed the SSL Cert and Key
+KEEP_NAMESPACE | false | Always keep the APB namespace after execution.
+KEEP_NAMESPACE_ON_ERROR | true | Keeps the APB namespace after a error during execution.
```
Comment (jmrodri) [inline docs/deployment.md]: TYPO: a error -> an error

File: etc/example-config.yaml [hunk 3]
```diff
@@ -33,6 +33,8 @@ openshift:
   bearer_token_file: ""
   image_pull_policy: IfNotPresent
   sandbox_role: "edit"
+  keep_namespace: false
+  keep_namespace_on_error: false
```
Comment (jmrodri) [inline etc/example-config.yaml]: the deployment above shows `keep_namespace_on_error = true`, but in the example we default it to false.

File: pkg/apb/svc_acct.go [hunk 4]
```diff
@@ -279,6 +287,19 @@ func (s *ServiceAccountManager) DestroyApbSandbox(executionContext ExecutionCont
 	return nil
 }
 
+func shouldDeleteNamspace(clusterConfig ClusterConfig, pod *apicorev1.Pod, getPodErr error) bool {
```
Comment (jmrodri) [inline pkg/apb/svc_acct.go]: TYPO: `shouldDeleteNamspace` -> `shouldDeleteNamespace`

File: etc/example-config.yaml (hunk 3 above)
Comment (shawn-hurley) [reply etc/example-config.yaml]: the default above needs to be false

**Comment by shawn-hurley** [comment]
This needs to wait for the #464 to merge and then this implements that PR. 
* if changes are made to 464 changes will need to be made to this PR.

**Pull request merged**